libs/domain/picking/src/components/picking-product-card/picking-product-card.component.ts

```diff
@@ -0,0 +1,184 @@
+  static styles = styles;
+
+  @property() productItem?: PickingListItem;
+  @property() status?: string;
+
+  @state() isCorrectNumberOfPickedProvided = true;
+  @state() currentNumberOfPicked?: number;
+  @state() pickedDataEvent?: ProductItemPickedEvent;
+
+  protected summaryInfo: SummaryInfo | undefined;
+
+  protected onSubmit(e: SubmitEvent): void {
+    e.preventDefault();
+
+    this.currentNumberOfPicked =
+      this.currentNumberOfPicked ?? this.productItem?.numberOfPicked;
+
+    if (this.productItem?.product.id) {
+      this.dispatchPickingEvents(EVENT_SUBMIT, {
+        productId: this.productItem.product.id,
+        numberOfPicked: this.currentNumberOfPicked,
+      });
+    }
+  }
+
+  protected editProductPicking(): void {
+    this.dispatchPickingEvents(EVENT_EDIT, {
+      productId: this.productItem?.product.id,
+    } as ProductItemPickedEvent);
+  }
+
+  protected onChangeQuantity({ detail: { quantity } }: CustomEvent): void {
+    this.currentNumberOfPicked = quantity;
+  }
+
+  protected getSummaryInfo(): SummaryInfo {
+    if (!this.productItem) {
+      return {
+        main: '',
+      };
+    }
+
+    if (this.status === ItemsFilters.Picked) {
+      const template = `${this.productItem.numberOfPicked}/${this.productItem.quantity}`;
+
+      if (this.productItem.numberOfPicked < this.productItem.quantity) {
+        return {
+          main: `${template} ${i18n('picking.product-card.items-picked')}`,
+        };
+      }
+
+      return {
+        main: template,
+        additional: `${i18n('picking.product-card.all-items-picked')}`,
+      };
+    } else if (this.status === ItemsFilters.NotFound) {
+      const template = `${this.productItem.numberOfNotPicked}/${this.productItem.quantity}`;
+
+      if (this.productItem.numberOfNotPicked < this.productItem.quantity) {
+        return {
+          main: `${template} ${i18n('picking.product-card.items-not-found')}`,
+        };
+      }
+
+      return {
+        main: template,
+        additional: `${i18n('picking.product-card.no-items-found')}`,
+      };
+    }
+
+    return {
+      main: '',
+    };
+  }
+
+  protected dispatchPickingEvents(
+    event: string,
+    productItem?: ProductItemPickedEvent
+  ): void {
+    this.dispatchEvent(
+      new CustomEvent(event, {
+        bubbles: true,
+        composed: true,
+        detail: productItem,
+      })
+    );
+  }
+
+  protected override render(): TemplateResult {
+    return html`${this.renderPickingProduct()}`;
+  }
+
+  protected renderPickingProduct(): TemplateResult {
+    if (this.productItem && this.currentNumberOfPicked) {
+      this.isCorrectNumberOfPickedProvided =
+        0 <= this.currentNumberOfPicked &&
+        this.currentNumberOfPicked <= this.productItem.quantity;
+    }
+
+    return html`
+      ${when(
+        this.productItem,
+        () => html`
+          <oryx-card>
+            <oryx-heading slot="heading" class="title">
+              ${this.productItem?.orderItem?.name}
```

Comment:
`when` creates one more nested level and requires conditional chains.

Reply:
Sorry not getting this one, the if is also a conditional, using 'when' I would say it's more lit native since this is essentially a wrapper around a ternary expression. 
We don't see many if statements inside our html templates

Reply:
You can use `if` statement with return because you haven’t `else` state. Benefit of using is better readability of code because you shouldn’t scroll down to see if it is `else` statement and there are less nesting lines. 
Also when you will use `if` statement then you can avoid conditional chains (`this.productItem?.`)